### PR TITLE
port devicelab from idevice_id -> xcdevices

### PR DESCRIPTION
**Pull request by christopherfujino**
## Description

The use of idevice_id from the tool has already be deprecated. However, the devicelab is still using `idevice_id` for device discovery. This PR instead calls `flutter devices --machine` to get the list devices, leveraging the existing tool code.

## Related Issues

Unblocks https://github.com/flutter/flutter/issues/50299.

## Tests

I manually ran iOS devicelab tests locally. Ultimately, post-submit will be the integration test for this change.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

### CODE REVIEW

**Review by jmagman — APPROVED** [review]
LGTM

File: dev/devicelab/lib/framework/adb.dart [hunk 1]
```diff
@@ -551,29 +551,49 @@ class IosDeviceDiscovery implements DeviceDiscovery {
...
   Future<List<String>> discoverDevices() async {
-    final String ideviceIdPath = path.join(getArtifactPath(), 'libimobiledevice', 'idevice_id');
-    final List<String> iosDeviceIDs = LineSplitter.split(await eval(ideviceIdPath, <String>['-l'], environment: _ideviceIdEnvironment))
-      .map<String>((String line) => line.trim())
-      .where((String line) => line.isNotEmpty)
-      .toList();
-    if (iosDeviceIDs.isEmpty)
+    final List<dynamic> results = json.decode(await eval(
+      path.join(flutterDirectory.path, 'bin', 'flutter'),
+      <String>['devices', '--machine', '--suppress-analytics'],
```
Comment (jonahwilliams) [inline dev/devicelab/lib/framework/adb.dart:558]: I would also pass --no-version-check just to be safe
https://github.com/flutter/flutter/blob/75fd73d81888966746821e65d37e1fc90dc56bfb/packages/flutter_tools/lib/src/runner/flutter_command_runner.dart#L82

Comment (christopherfujino) [reply dev/devicelab/lib/framework/adb.dart:558]: Sorry, landed this while the tree was green. Fortunately, in the command runner we bypass the version check if `--machine` is provided https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command_runner.dart#L82

**Comment by flutter-github-sync** [comment]
Google testing passed!

**Pull request merged**